scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -956,6 +956,33 @@ class Router(formatOps: FormatOps) {
             .onlyIf(open.pos.endLine == close.pos.startLine)
             .withSingleLine(close, killOnFail = true)
         ) ++ oneArgPerLineSplits
+
+      case tok @ FormatToken(T.LeftArrow(), right, _)
+          if leftOwner.is[Enumerator.Generator] =>
+        val lastToken = leftOwner.tokens.last
+        val indent: Length =
+          if (style.align.arrowEnumeratorGenerator) StateColumn
+          else if (right.is[T.LeftBrace]) Num(-2)
```

Comment:
this looks unusual. what does this do?

Reply:
The same as https://github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala#L376-L378

Formatting without `-2`:
```scala
xs.map {
  case x => {
      println("x")
    }
}

for {
  x <- {
       println("x")
     }
} yield x
```

Reply:
ah, thank you, so that's why it was there in that case statement. always puzzled me.